src/main/msp/msp_serial.c

```diff
@@ -447,11 +447,20 @@ static mspPostProcessFnPtr mspSerialProcessReceivedCommand(mspPort_t *msp, mspPr
 
 static void mspEvaluateNonMspData(mspPort_t * mspPort, uint8_t receivedChar)
 {
-   if (receivedChar == serialConfig()->reboot_character) {
+    if (receivedChar == serialConfig()->reboot_character) {
         mspPort->pendingRequest = MSP_PENDING_BOOTLOADER_ROM;
 #ifdef USE_CLI
-   } else if (receivedChar == '#') {
-        mspPort->pendingRequest = MSP_PENDING_CLI;
+    } else {
+        switch(receivedChar) {
+            case '#':
+                mspPort->pendingRequest = MSP_PENDING_CLI;
```

Comment:
This has always actually been broken, and in essence makes the guard time in effective. Will need to think about the logic in mspSerialProcess, such that there is perhaps a new state "pending request".

I also think we handle non MSP chars first, if the state is IDLE, and then pass to MSP processing instead of the current processing flow.

It might be better to split the states into port state, versus msp state.